### ISSUE TITLE
Export to CSV

### ISSUE THREAD

**Issue by TriPSs** [TriPSs/nestjs-query#305]
**Is your feature request related to a problem? Please describe.**
When exporting data we now have to create endpoints for all the data types, would be easier if we, just like aggregation, can enable this.

### Have you read the [Contributing Guidelines](https://github.com/tripss/nestjs-query/blob/master/CONTRIBUTING.md)?

Yes

**Describe the solution you'd like**
a `<DTO>Export` method that accepts the same filters and additionally a map for the columns `{ field: string, headerName: string }` (if not provided use field names) and `format = ExportFormat` where `ExportFormat` only accepts `CSV` for now.

**Describe alternatives you've considered**
Creating a service/resolver class for each data type ourselfs.

**Additional context**

**Comment by manlyman29** [comment]
Is there any ETA or active work going on for this? I was actually looking for this.

**Comment by TriPSs** [comment]
Not yet, still want it but did not have any time for it yet.

**Comment by TriPSs** [comment]
@lume-code yes, you could copy the read resolver and remove the pagina from it, then everything that works in query many will work with the CSV export.  Use it to fetch the data and then create the CSV (based off the `format = ExportFormat`).

**Comment by lume-code** [comment]
@TriPSs Great, Thanks, I will give it a shot😊

**Comment by lume-code** [comment]
This was my attempt at an export resolver that works for most parts except for the relation retrieval part:

https://gist.github.com/lume-code/5a0e90a2b6526e4a23ed98c11335b445

**Comment by TriPSs** [comment]
Closing as part of repository cleanup. If this issue is still relevant, please reopen.

PS: Once the `query-rest` package is live you can use the export controller of it.

**Comment by TriPSs** [comment]
@TriPSs you indeed could, but would loose all the GraphQL filters.

**Comment by TriPSs** [comment]
This is now available in the alpha of v11
